docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,337 @@
+
+There are a few more configurable options of lesser interest:
+- `recommenderInterval` (default 1m): How often VPA retrieves the pods and metrics respectively how often it recomputes the recommendations and bounds.
+
+There are many more options that you can only configure if you deploy your own VPA and which we will not discuss here, but you can check them out [here](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/FAQ.md#what-are-the-parameters-to-vpa-recommender).
+
+### VPA Updater Options
+
+You can read up the [full VPA updater options](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/FAQ.md#what-are-the-parameters-to-vpa-updater) online and set some of them conveniently in your [Gardener shoot cluster spec](https://github.com/gardener/gardener/blob/957e4c7/example/90-shoot.yaml#L298-L307):
+- `evictAfterOOMThreshold` (default 10m): Pods where at least one container OOMs within this time period since its start will be actively evicted, which will implicitly apply the new target recommendation that will have been [bumped up after `OOMKill`](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#custom-memory-bump-up-after-oomkill). Please note, the kubelet may evict pods even before an OOM, but only if `kube-reserved` is underrun, i.e. [node-level resources are running low](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#pod-selection-for-kubelet-eviction). In these cases, eviction will happen first by pod priority and second by how much the usage overruns the requests.
+- `evictionTolerance` (default 50%): Defines a threshold below which no further eligible pod will be evited anymore, i.e. limits how many eligible pod may be in eviction in parallel (but at least 1). The [threshold is computed as follows](https://github.com/kubernetes/autoscaler/blob/4d0511363eeeff657119797dd8d26e851dcc3459/vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go#L108-L117): `running - evicted > replicas - tolerance`. Example: 10 replicas, 9 running, 8 eligible for eviction, 20% tolerance with 10 replicas which amounts to 2 pods, and no pod evicted in this round yet, then `9 - 0 > 10 - 2` is true and a pod would be evicted, but the next one would be in violation as `9 - 1 = 10 - 2` and no further pod would be evicted anymore in this round.
```

Comment:
I wouldn't mention this at all and just direct people to PDB. Do we or others have a concrete use-case for this, such that we need to put `evictionTolerance`, `evictionRateBurst` and `evictionRateLimit` here? Maybe because of DaemonSets, which don't have a PDB? If so, we should probably mention this more directly.

Reply:
See above, same argument: we have exposed it in Gardener and I feel pressed to explain it before people jump to the wrong conclusions. We also know, that PDBs are no protection (see the ETCD case).

I added a warning and the daemonset comment:
> In general, avoid modifying these eviction settings unless you have good reasons and try to rely on [Pod Disruption Budgets (PDBs)](https://kubernetes.io/docs/tasks/run-application/configure-pdb) instead. However, [PDBs are not available for daemon sets](https://github.com/kubernetes/kubernetes/issues/108124).